schemas/pa-yaml/v3.0/pa.schema.yaml

```diff
@@ -401,24 +401,41 @@ definitions:
   DataSource-name:
     $ref: "#/definitions/entity-name"
 
-  DataSource-Type:
-    enum:
-      - DataverseTable
-
   DataSource-instance:
     type: object
-    required: [Type]
+    required:
+      - Type
+    additionalProperties: false
     properties:
-      Type: { $ref: "#/definitions/DataSource-Type" }
+      Type:
+        type: string
+        enum:
+          - Table
+          - Actions
+      ConnectorId:
+        type: string
+        nullable: true
```

Comment:
sorry I missed this. I don't think 'nullable' is valid in json schema. I've noticed that sometimes the vs code redhat ext suggests patterns in the json schema that's not valid.

For what you want to validate, it's best to just state that it should be a string, and never null.